implementations/rust/ockam/ockam_transport_tcp/src/portal/inlet_listener.rs

```diff
@@ -1,14 +1,20 @@
 use crate::portal::addresses::{Addresses, PortalType};
-use crate::{portal::TcpPortalWorker, HostnamePort, TcpInlet, TcpInletOptions, TcpRegistry};
+use crate::{portal::TcpPortalWorker, TcpInlet, TcpInletOptions, TcpRegistry};
 use ockam_core::compat::net::SocketAddr;
 use ockam_core::compat::sync::{Arc, RwLock};
 use ockam_core::{async_trait, compat::boxed::Box};
 use ockam_core::{Address, Processor, Result, Route};
 use ockam_node::Context;
-use ockam_transport_core::TransportError;
+use ockam_transport_core::{HostnamePort, TransportError};
 use tokio::net::TcpListener;
 use tracing::{debug, error, instrument};
 
+#[derive(Debug, Clone)]
+pub struct OutletSharedState {
```

Comment:
Between `TcpInletListenProcessor` and `TcpInlet`. Maybe there is a confusion with naming because it's not really Outlet's state. It was information inlet knows about the outlet, before I added is_paused it did make some sense. Now I think it should be `InletSharedState`?